bugbug/tools/code_review.py

```diff
@@ -1116,15 +1116,25 @@ def find_similar_patch_comments(self, patch: Patch, limit: int):
 
         patch_set = PatchSet.from_string(patch.raw_diff)
 
-        yielded = 0
+        # We want to avoid returning the same comment multiple times. Thus, if
+        # a comment matches multiple hunks, we will only consider it once.
+        max_score_per_comment = {}
         for patched_file in patch_set:
             if not patched_file.is_modified_file:
                 continue
 
             for hunk in patched_file:
                 for result in self.find_similar_hunk_comments(hunk):
-                    yield result
+                    if (
+                        result.id not in max_score_per_comment
+                        or result.score > max_score_per_comment[result.id]
+                    ):
+                        max_score_per_comment[result.id] = result
+
+        results_with_score = sorted(
+            (result.score, result)
+            for result in max_score_per_comment.values()
+        )
+        list_found = [e[1] for e in results_with_score[-limit:]]
```

Comment:
nit: this will be self-explanatory to improve the readability:
```suggestion

        return [result for score, result in results_with_score[-limit:]]
```